mne/encoding/model.py

```diff
@@ -1,28 +1,66 @@
 import numpy as np
 from sklearn.base import BaseEstimator
 from scipy import linalg
+from .feature import EventsBinarizer, DataDelayer
 from ..externals.six import string_types
+from ..utils import warn
+from sklearn.base import is_regressor
+
+
+class EventRelatedRegressor(object):
+    def __init__(self, raw, events, est=None, event_id=None, tmin=-.5, tmax=.5,
+                 preproc_x=None, preproc_y=None):
+        if events.shape[-1] != 3:
+            raise ValueError('Events must be shape (n_events, 3)')
+        if raw.preloaded is False:
+            raise ValueError('Data must be preloaded')
+
+        # Create events representation
+        self.ev_ixs = events[:, 0]
+        self.ev_types = events[:, 2]
+        self.event_id = event_id
+        binarizer = EventsBinarizer(raw.n_times, sfreq=raw.info['sfreq'])
+        self.X = binarizer.fit_transform(self.ev_ixs, self.ev_types,
+                                         self.event_id)
```

Comment:
sklearn objects don't store `X`. Maybe use explicit naming instead of X here. You'll pass it as an X when you give it to a `fit` method.